src/Compilers/VisualBasic/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.vb

```diff
@@ -0,0 +1,303 @@
+            Return AddDynamicAnalysis(original, MyBase.InstrumentForEachLoopInitialization(original, initialization))
+        End Function
+
+        Public Overrides Function InstrumentForLoopInitialization(original As BoundForToStatement, initialization As BoundStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.InstrumentForLoopInitialization(original, initialization))
+        End Function
+
+        Public Overrides Function InstrumentLocalInitialization(original As BoundLocalDeclaration, rewritten As BoundStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.InstrumentLocalInitialization(original, rewritten))
+        End Function
+
+        Public Overrides Function CreateUsingStatementPrologue(original As BoundUsingStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.CreateUsingStatementPrologue(original))
+        End Function
+
+        Public Overrides Function CreateWithStatementPrologue(original As BoundWithStatement) As BoundStatement
+            Return AddDynamicAnalysis(original, MyBase.CreateWithStatementPrologue(original))
+        End Function
+
+        Private Function AddDynamicAnalysis(original As BoundStatement, rewritten As BoundStatement) As BoundStatement
+            If Not original.WasCompilerGenerated Then
+                Return CollectDynamicAnalysis(original, rewritten)
+            End If
+
+            Return rewritten
+        End Function
+
+        Private Function CollectDynamicAnalysis(original As BoundStatement, rewritten As BoundStatement) As BoundStatement
+            Dim statementFactory As New SyntheticBoundNodeFactory(_factory.TopLevelMethod, _method, original.Syntax, _factory.CompilationState, _diagnostics)
+            Dim analysisPoint As BoundStatement = AddAnalysisPoint(SyntaxForSpan(original), statementFactory)
+            Return If(rewritten IsNot Nothing, statementFactory.StatementList(analysisPoint, rewritten), analysisPoint)
+        End Function
+
+        Private Function AddAnalysisPoint(syntaxForSpan As VisualBasicSyntaxNode, statementFactory As SyntheticBoundNodeFactory) As BoundStatement
+            ' Add an entry in the spans array.
+
+            Dim location As Location = syntaxForSpan.GetLocation()
+            Dim spanPosition As FileLinePositionSpan = location.GetMappedLineSpan()
+            Dim path As String = spanPosition.Path
+            If path.Length = 0 Then
+                path = syntaxForSpan.SyntaxTree.FilePath
+            End If
+
+            Dim spansIndex As Integer = _spansBuilder.Count
+            _spansBuilder.Add(New SourceSpan(_debugDocumentProvider.Invoke(path, ""), spanPosition.StartLinePosition.Line, spanPosition.StartLinePosition.Character, spanPosition.EndLinePosition.Line, spanPosition.EndLinePosition.Character))
```

Comment:
OK on using the named parameter. As for the value, I have no idea. I copied this code from an example. @tmat can provide a definitive answer.